sw/host/spiflash/ftdi_spi_interface.cc

```diff
@@ -106,22 +109,82 @@ bool FtdiSpiInterface::TransmitFrame(const uint8_t *tx, uint8_t *rx,
     std::cerr << "Unable to start spi transaction." << std::endl;
     return false;
   }
+
   uint8_t *tmp_rx = ::Transfer(spi_->ctx, tx_local.data(), size);
+  // We're not using the result of this read, so free it right away.
+  if (tmp_rx) {
+    free(tmp_rx);
+  }
+
   if (Stop(spi_->ctx)) {
     std::cerr << "Unable to terminate spi transaction." << std::endl;
-    if (tmp_rx) {
-      free(tmp_rx);
-    }
     return false;
   }
   if (tmp_rx == nullptr) {
     std::cerr << "Unable to transfer data. Frame size: " << size << std::endl;
     return false;
   }
-  usleep(kTransmitDelay);
-  memcpy(rx, tmp_rx, size);
-  free(tmp_rx);
   return true;
 }
+
+bool FtdiSpiInterface::CheckHash(const uint8_t *tx, size_t size) {
```

Comment:
I think this works because the BootRom will not increment the expected frame counter until it sees a valid header. So SGTM at first glance.